Userland/Libraries/LibWeb/CSS/StyleValues/CalculatedStyleValue.cpp

```diff
@@ -390,6 +390,106 @@ ErrorOr<void> InvertCalculationNode::dump(StringBuilder& builder, int indent) co
     return {};
 }
 
+ErrorOr<NonnullOwnPtr<MinCalculationNode>> MinCalculationNode::create(Vector<NonnullOwnPtr<Web::CSS::CalculationNode>> value)
+{
+    return adopt_nonnull_own_or_enomem(new (nothrow) MinCalculationNode(move(value)));
+}
+
+MinCalculationNode::MinCalculationNode(Vector<NonnullOwnPtr<CalculationNode>> value)
+    : CalculationNode(Type::Invert)
+    , m_values(move(value))
+{
+}
+
+MinCalculationNode::~MinCalculationNode() = default;
+
+ErrorOr<String> MinCalculationNode::to_string() const
+{
+    StringBuilder builder;
+    builder.append("min("sv);
+    for (size_t i = 0; i < m_values.size(); ++i) {
+        if (i != 0)
+            builder.append(", "sv);
+        builder.append(TRY(m_values[i]->to_string()));
+    }
+    builder.append(")"sv);
+    return builder.to_string();
+}
+
+Optional<CalculatedStyleValue::ResolvedType> MinCalculationNode::resolved_type() const
+{
+    CalculatedStyleValue::ResolvedType type;
+    bool has_first = false;
+    for (auto const& value : m_values) {
+        auto value_type = value->resolved_type();
+        if (!value_type.has_value())
+            return {};
+
+        auto type_value = value_type.value();
+
+        if (!has_first) {
+            has_first = true;
+            type = type_value;
+        }
+
+        if (type_value != type)
+            return {};
+    }
+
+    return type;
+}
+
+bool MinCalculationNode::contains_percentage() const
+{
+    for (auto const& value : m_values) {
+        if (value->contains_percentage())
+            return true;
+    }
+
+    return false;
+}
+
+CalculatedStyleValue::CalculationResult MinCalculationNode::resolve(Layout::Node const* layout_node, CalculatedStyleValue::PercentageBasis const& percentage_basis) const
+{
+    auto smallest_value = NumericLimits<float>::max();
+    CalculatedStyleValue::CalculationResult smallest_node = m_values.first()->resolve(layout_node, percentage_basis);
```

Comment:
`smallest_value` should be the value of `smallest_node`, otherwise the first node will never get chosen.